### DISCUSSION TITLE
Would it be a good idea to apply Apache Arrow for .net?

### DISCUSSION THREAD

**Discussion by ingted** [NTDLS/Katzebase#58] [Ideas]
If it supports storing data in the Arrow format,, maybe it would attract more users and contributors?

**Comment by NTDLS** [comment]
The current data is stored in compressed protobuf pages of hashed string value pairs which are "materialized" on demand. These pages are used for concurrency, durability and atomic operations. They allow for locking with configurable levels of granular lock escalation. The pages also allow for ephemeral memory paging, delayed write (within transactions) and external b-tree (albeit custom) indexing.

Unfortunately the key-value-pairs are integral to the engine.

**Reply by NTDLS** [reply]
Also, sorry for the late reply. I with I could get more "in my face" notifications from GitHub.
